### ISSUE TITLE
TF-DF Compatibility with Keras 3?

### ISSUE THREAD

**Issue by rlcauvin** [tensorflow/decision-forests#210]
According to the [known issues](https://www.tensorflow.org/decision_forests/known_issues), "TF-DF is not compatible with Keras 3 at this time." Are efforts underway to update TF-DF to be compatible with Keras 3?

**Comment by rstz** [comment]
Yes, we are planning to make TF-DF compatible with Keras 3, but there is no timeline yet, since it might be a non-trivial change.

**Comment by woodwj** [comment]
Ive been trying and failing to tune hyperameters for tfdf.CartModels in line with the google developer course practice https://developers.google.com/machine-learning/decision-forests/practice

but i get a FatalTypeError after the keras_tuner builds the hypermodel as its not a keras.model.Model instance.

My guess is that now keras_tuner uses keras 3 (Release v1.4.6), it doesnt recognise these as keras modes?

Heres a notebook replicating the issue
https://colab.research.google.com/drive/183uPNtxGXn7TI8c9CDbrJ3sD0KWr-3ms#scrollTo=uO8SKW48MVPS

**Comment by rstz** [comment]
Hi, looks like we have to update this class - thank you for notifying us. Very likely, you only need to add
```
import os
# Keep using Keras 2
os.environ['TF_USE_LEGACY_KERAS'] = '1'
```
at the top of the colab (note that I cannot access yours, since it's not shared with the world).

**Comment by woodwj** [comment]
Fixed with versioning keras_tuner AND setting legacy env variable. Il share this feedback this the course.

Im not really familiar with keras, but maybe the backward compatibility introduced here doesn't get the right backend? im not quite sure where to open a bug report, is a keras_tuner or tfdf issue?
keras_tuner 1.4.5 -> 1.4.6 [https://github.com/keras-team/keras-tuner/compare/v1.4.5...v1.4.6]

`
import os
# Keep using Keras 2
os.environ['TF_USE_LEGACY_KERAS'] = '1'
# use 
!pip install keras_tuner==1.4.5
`
